### PR TITLE
Initial migration of api logic to geometry

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: api/shape/Arc.js [hunk 1]
```diff
@@ -97,7 +98,7 @@ const reifyArc =
       }
     }
 
-    return spiral.absolute();
+    return absolute()(spiral);
```
Comment (BarbourSmith) [inline api/shape/Arc.js:101]: How does this syntax work?

File: api/shape/Arc.js [hunk 2]
```diff
@@ -160,26 +161,27 @@ const ArcOp =
...
     const [c1, c2] = await buildCorners(x, y, z)(null);
-    const result = reify({ c1, c2, start, end, sides, zag });
-    return result;
+    const result = await reify({ c1, c2, start, end, sides, zag });
+    const geometry = await result.toGeometry();
+    return geometry;
   };
 
-export const Arc = Shape.registerMethod2(
+export const Arc = Shape.registerMethod3(
```
Comment (BarbourSmith) [inline api/shape/Arc.js:169]: Three! 😃

File: api/shape/Curve.js [hunk 3]
```diff
@@ -7,15 +7,19 @@ import Shape from './Shape.js';
...
+  ) => {
     const interpolator = new CurveInterpolator(coordinates, {
-      closed: isClosed,
+      closed,
       tension: 0.2,
       alpha: 0.5,
     });
     const points = interpolator.getPoints(steps);
-    if (isClosed) {
+    if (closed) {
```
Comment (BarbourSmith) [inline api/shape/Curve.js:22]: I like this name change 👍

File: api/shape/asPart.js [hunk 4]
```diff
@@ -1,12 +1,11 @@
 import Shape from './Shape.js';
-import { taggedItem } from '@jsxcad/geometry';
+import { asPart as op } from '@jsxcad/geometry';
```
Comment (BarbourSmith) [inline api/shape/asPart.js:2]: Why the switch to all of these being 'op' now? Just more condensed?

File: api/shape/view.js [hunk 5]
```diff
@@ -14,22 +14,22 @@ const MODES =
   'modes:grid,none,side,top,wireframe,noWireframe,skin,noSkin,outline,noOutline';
 
 const applyModes = async (shape, options, modes) => {
-  if (modes.includes('wireframe')) {
+  if (modes.wireframe) {
```
Comment (BarbourSmith) [inline api/shape/view.js:17]: This is nicer and cleaner 😍

File: nb/api/Icosahedron.md [hunk 6]
```diff
@@ -8,7 +8,7 @@ Produces an icosahedron that fits the bounding dimensions.
 Icosahedron(3)
 
 ```JavaScript
-Icosahedron(3).and(Box(3, 3, 3).material('glass')).view().md('Icosahedron(3)');
+Icosahedron(3).and(Box(3, 3, 3).material('glass')).view().note('Icosahedron(3)');
```
Comment (BarbourSmith) [inline nb/api/Icosahedron.md:11]: 👍

File: api/shape/Arc.js (hunk 2 above)
Comment (pentacular) [reply api/shape/Arc.js:169]: Yeah, once everything is up to 3, I think it'll be time to remove the numbers. :)

File: api/shape/Arc.js (hunk 1 above)
Comment (pentacular) [reply api/shape/Arc.js:101]: At the api/shape level, operators are functors.

f = op(args) builds a function which waits for an input shape for context.
r = f(input) then calls it.

This allows us to pass around things like cut(1/4) and then later choose what shape to apply it to.

Here we're just saying, call absolute with no parameters on spiral.

**Pull request merged**